src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -73,6 +72,13 @@ private void NodeView_MouseLeave(object sender, MouseEventArgs e)
                 viewModel.OnMouseLeave();
         }
 
+        /// <summary>
+        /// This ensures that a preview control is initialized only when it is absolutely
+        /// needed. It is static so that it can be shared for all node views, instead of
+        /// each node having its own debouncer with additional overhead.
+        /// </summary>
+        private static ActionDebouncer delayPreviewControl = null;
```

Comment:
An alternative to having a static that you need to manage for every WorkspaceChange would be to store the `delayPreviewControl` debouncer as an internal (non-static) member inside the WorkspaceViewModel class.
That way you can better control the life cycle (initialize on workspace constructor and cleanup on dispose).
NodeView has access to the workspace view model (ex. `viewModel?.WorkspaceViewModel?.delayPreviewControl?.Debounce`)

Reply:
Great idea, [it worked](https://github.com/dimven-adsk/Dynamo/commit/c1606be3aa295a422ed9c100974d27378d124393) just as well and produced better structured code.